tools/cli.js

```diff
@@ -24,9 +24,17 @@ const log = (msg) => {
   if (userInput) rl.write(userInput);
 };
 
-function complete(input, completions) {
-  if (!input) return completions;
-  return completions.filter(c => c.startsWith(input));
+function complete(input, completions, forbidden = []) {
+  if (!forbidden && !input) return completions;
+  return completions.filter(
```

Comment:
As far as I understand, the result of this `filter` will always have finite number of possible values on every successful connection with interface inspection, and it does not depend on any user input after connection, which means it can be optimized by computing possible values on connection establishment (after inspection) and using them afterwards.